### ISSUE TITLE
Running db pull removes name from @@id

### ISSUE THREAD

**Issue by itsUndefined** [prisma/prisma#14556]
### Bug description

Similar old issue I found: https://github.com/prisma/prisma/issues/9318
This time, the name of the id gets lost on db pull

Expected Result:
`@@id([userId, companyId], name: "id")`

Actual Result:
`@@id([userId, companyId])`

### How to reproduce

- Create a schema and add @@id with a specific name
- Push all changes to db with `prisma db push`
- Run `prisma db pull`

### Expected behavior

The name should be retained:
`@@id([userId, companyId], name: "id")`

### Prisma information

```
model User { 
    userId         Int    @default(autoincrement()) @map("id")
    companyId Int 

    @@id([userId, companyId], name: "id")
}
```

### Environment & setup

- OS: Windows 11
- Database: MariaDB
- Node.js version: 16.15.1

### Prisma Version

```
prisma                  : 4.1.1
@prisma/client          : 4.1.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\migration-engine-windows.exe)        
Introspection Engine    : introspection-core 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\introspection-engine-windows.exe)      
Format Binary           : prisma-fmt 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 8d8414deb360336e4698a65aa45a1fbaf1ce13d8
Studio                  : 0.469.0
Preview Features        : interactiveTransactions
```

**Comment by janpio** [comment]
What is the SQL of `User`? That would make it easier to reproduce this properly. Thanks.

**Comment by tomhoule** [comment]
> What is the SQL of User? That would make it easier to reproduce this properly. Thanks.

`name` has nothing corresponding to it in the SQL definition of the primary key, this wouldn't help. We only need to reproduce that the name argument isn't reintrospected to confirm.

**Comment by itsUndefined** [comment]
Sorry for the long delay to respond. I created a new project and confirmed that the following schema file is enough to reproduce the issue on a brand new project:

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}


model User { 
    userId         Int    @default(autoincrement()) @map("id")
    companyId Int 

    @@id([userId, companyId], name: "id")
}
```

**Comment by tomhoule** [comment]
Thanks!

**Comment by pimeys** [comment]
Fixed in commit: https://github.com/prisma/prisma-engines/commit/bffd935029e462592819b33a080e8f8953b58acb

Should be fixed in Prisma version 4.6.0.
